packages/context/src/context.ts

```diff
@@ -53,6 +53,15 @@ import {iterator, multiple} from 'p-event';
 
 const debug = debugFactory('loopback:context');
 
+/**
+ * Synchronous event listener for the `Context` as en event emitter
+ */
+export type ContextEventListener = (
+  binding: Readonly<Binding<unknown>>, // Binding to be added or removed
+  context: Context, // The source context
+  event: string, // 'bind' or 'unbind'
+) => void;
```

Comment:
Positional arguments are not very easy to use when the number of arguments is more than 1-2, plus they are difficult to extend with optional arguments in the future. Can we rework the signature to use an object with named properties please?

```ts
export interface ContextEvent {
  binding: Readonly<Binding<unknown>>; // Binding to be added or removed
 context: Context; // The source context
 event: string; // 'bind' or 'unbind'
}

export type ContextEventListener = (event: ContextEvent) => void;
```

We can use object de-structuring to keep usage easy:

```ts
function myListener({binding, context, event}: ContextEvent) {
  // ...
}
```